### ISSUE TITLE
Queries about training

### ISSUE THREAD

**Issue by nishantb21** [ahmetgunduz/Real-time-GesRec#24]
Hello,

I am trying to implement your paper from scratch as part of my project and have some questions which I was hoping you could answer. I am only trying to train the detector half of the network for now and using the JESTER dataset to do so.

1. How is the data getting fed in? Each folder has 'n' frames which belong to a category(i.e. gesture or no gesture). Taking the detector queue as 8 frames, do you then split the 'n' frames into n/8 chunks each having the gesture or no gesture label?
2. How long did you pre-train it on JESTER for? You paper mentions 25 epochs but I am guessing that'd for the classifier? Your code seems to indicate 100 epochs instead.

I am hoping you can help me with this. Thanks in advance!

Regards,
Nishant Bhattacharya

**Comment by okankop** [comment]
First of all, detectors are not pretrained on Jester dataset, only the classifiers are pretrained on Jester. Also, Jester dataset does not contain continuous videos, meaning that there is no "no-gesture" part.

1. For selecting n frames, TemporalRandomCrop is applied selecting a consecutive n frames at random positions from a no-gesture/gesture part.

2. Classifiers are trained on Jester for only 25 epochs.

Hope these answer your questions!

**Comment by nishantb21** [comment]
Apologies for the dumb questions but I am still unable to get a clear picture as to how the training for the detector happens. 

1. Am I correct in assuming that the detector is trained separately first and then frozen after which the classifier is trained?
2. If my first assumption is correct, how long do you train the detector for? You mentioned that you don't train the detector on JESTER. So do you train it directly on nvGesture and EgoGesture dataset? 
3. Am I correct in assuming that the data going into the detector for training data is of the form [frame 1, frame 2, .... , frame n] and the label is either [1, 0] or [0, 1] depending on whether the video has gesture or not?

Thank you so much!

**Comment by okankop** [comment]
1. The detector and classifier are trained offline and independent to each other. 
2. Yes, the detector is trained directly on nvGesture and EgoGesture datasets.
3. You are correct.

**Comment by nishantb21** [comment]
Thanks for all your help!
